src/IrcCommands/Commands/Invite.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <list>
+#include "../../Channel/Channel.hpp"
+#include "../../Client/Client.hpp"
+#include "../../Server/Server.hpp"
+#include "../../debug.hpp"
+#include "../../includes/CONSTANTS.hpp"
+#include "../../includes/types.hpp"
+#include "../IrcCommands.hpp"
+
+/**
+ * @brief INVITE command is used to invite clients to channels
+ * Only members of that channel can invite non-members
+ * If invite_only (+i) mode is enabled for the channel, only chanops
+ * can successfully call the command.
+ * After a successful call invited client is getting added to _invited list
+ * of the channel. Caller client receives RPL_INVITING and invited client
+ * receives an custom INVITE message from the server.
+ * e.g INVITE <nickname> <channel>
+ * 
+ * TODO:
+ * (1) implement and send RPL_341 to caller after a successful call 
+ * (2) implement and send an INVITE message to the invited client
+ *  
+ * @return 0, in case of an error it returns error codes:
+ * RPL_INVITING (341)
+ * ERR_NEEDMOREPARAMS (461)
+ * ERR_NOSUCHCHANNEL (403)
+ * ERR_CHANOPRIVSNEEDED (482)
+ * ERR_USERNOTINCHANNEL (441)
+ * ERR_NOTONCHANNEL (442)
+ * 
+ * @return it returns 1 if command is successfully executed
+ */
+int IrcCommands::invite(Server& base, const struct cmd_obj& cmd) {
+  if (cmd.parameters.size() < 2) {
+    send_message(base, cmd, ERR_NEEDMOREPARAMS, true, NULL);
+    return (ERR_NEEDMOREPARAMS);
+  }
+  std::list<Channel>::iterator it_chan = base._channel_list.begin();
+  if (!base._channel_list.empty()) {
+    for (; it_chan != base._channel_list.end(); it_chan++) {
+      if (it_chan->get_name() == cmd.parameters[1])
+        break;
+    }
+  }
+  if (it_chan == base._channel_list.end() || base._channel_list.empty()) {
```

Comment:
to avoid segfaults I would out this the other waz around. 
```suggestion
  if (base._channel_list.empty() || it_chan == base._channel_list.end()) {
```